apps/src/calc/inputIterator.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Given a set of values (i.e. [1,2,3], and a number of parameters, generates
+ * all possible combinations of values.
+ */
+var InputIterator = function (values, numParams) {
```

Comment:
Moving over in general makes sense even given (a) just because it would put a Googl-able name to the process & I think reduce the chances of confusion/bug introduction. The density/readability _within_ the single `cartesianProduct` method doesn't quite bother me as much since it's a smaller, atomic operation we wouldn't need to / be able to modify.

It was definitely the right call to structure it this way before we knew there was a word for the operation, (they're all pretty tricky to parse the internals of).

Since this is a smaller utility, maybe a plan can be to move over to it next time (if ever) we need to modify that test input logic